src/components/day/Day.js

```diff
@@ -295,13 +295,43 @@ export default class DayComponent extends Field {
       this.addEventListener(this.refs.input, this.info.changeEvent, () => this.updateValue(null, {
         modified: true
       }));
+      this.addFocusBlurEvents();
     }
+    this.restoreFocus();
     this.setValue(this.dataValue);
     // Force the disabled state with getters and setters.
     this.disabled = this.shouldDisabled;
     return superAttach;
   }
 
+  addFocusBlurEvents() {
```

Comment:
It seems that instead of you fixing the underlying issue of the control being redrawn, instead you are saving the focused element and re-focusing when the element redraws.  Can we instead fix the underlying issue of why the component is redrawing to begin with?